src/main/java/io/pinecone/PineconeConfigs.java

```diff
@@ -0,0 +1,60 @@
+package io.pinecone;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PineconeConfigs {
+    private PineconeClientConfig clientConfig;
+    private PineconeConnectionConfig connectionConfig;
+
+    public PineconeConfigs() {
```

Comment:
I can initialize both config objects as a part of default constructor but if the 3 values are not passed, there's no need for new config objects since the user will instantiate both configs using .withClientConfig and .withConnectionConfig methods. But please validate this logic.